File: Src/Icm.Core/IO/TextReaderExtensions.cs
using System.Globalization;
using System.IO;

namespace Icm.IO
{
	public static class TextReaderExtensions
	{
		public static string Format(this TextReader tr, params object[] args)
		{
			var template = tr.ReadToEnd();
			tr.Close();
			return string.Format(CultureInfo.CurrentCulture, template, args);
		}

	}

}

File: Src2/Icm.Core/IO/TextReaderExtensions.vb
Imports System.Runtime.CompilerServices
Imports System.IO

Namespace Icm.IO

    Public Module TextReaderExtensions

        <Extension>
        Public Function Format(ByVal tr As TextReader, ByVal ParamArray args() As Object) As String
            Dim template = tr.ReadToEnd
            tr.Close()
            Return String.Format(CultureInfo.CurrentCulture, template, args)
        End Function

    End Module

End Namespace

